### ISSUE TITLE
[Semantic Segmentation]`iters_per_epoch` computed incorrectly in training script?

### ISSUE THREAD

**Issue by chenliu0831** [dmlc/gluon-cv#1079]
Looking at the input to LRScheduler from Semantic segmentation [training script](https://github.com/dmlc/gluon-cv/blob/master/scripts/segmentation/train.py#L151), shouldn't it be `len(self.train_data) // self.args.batch_size` which is the number of batches?

**Comment by bryanyzhu** [comment]
There are two concepts here, one is [trainset](https://github.com/dmlc/gluon-cv/blob/master/scripts/segmentation/train.py#L117), the other is [self.train_data](https://github.com/dmlc/gluon-cv/blob/master/scripts/segmentation/train.py#L121).

`trainset` contains all the image-label pairs in the dataset. It is a list.

`self.train_data` is the dataloader, which already divides the whole dataset into mini-batches by dividing `batch_size`. Its length is `len(trainset) // args.batch_size`. This is why `args.batch_size` shows up in its definition:

```
self.train_data = gluon.data.DataLoader(trainset, args.batch_size, ...)
```

So the input to the LRScheduler is correct. There is no need to divide `args.batch_size` again.

**Comment by chenliu0831** [comment]
@bryanyzhu Thanks for explanation. A quick followup, would calling `len(self.train_data)` load all training data into memory? If so that seems less ideally before the training starts.

**Comment by bryanyzhu** [comment]
No, it won't load all training data into memory. Its length is just an attribute, you can all it anytime.

**Comment by chenliu0831** [comment]
Thanks a lot for the explanation. Closing.
